File: bank-service/src/main/java/rs/edu/raf/repository/ExchangeInvoiceRepository.java
package rs.edu.raf.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import rs.edu.raf.model.ExchangeInvoice;

import java.util.List;

@Repository
public interface ExchangeInvoiceRepository extends JpaRepository<ExchangeInvoice,Long> {
    List<ExchangeInvoice> findAllBySenderCurrency_Code(String curr);
    List<ExchangeInvoice> findAllByReceiverCurrency_Code(String curr);
}

File: bank-service-kotlin/src/main/kotlin/org/example/bankservicekotlin/repository/ExchangeInvoiceRepository.kt
package org.example.bankservicekotlin.repository

import org.example.bankservicekotlin.model.ExchangeInvoice
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ExchangeInvoiceRepository: JpaRepository<ExchangeInvoice, Long> {
    fun findAllBySenderCurrency_Code(curr: String): List<ExchangeInvoice>
    fun findAllByReceiverCurrency_Code(curr: String): List<ExchangeInvoice>
}